### PR TITLE
opentrack: init at 2.3.13

### PR DESCRIPTION
**Pull request by zaninime**
###### Motivation for this change
Add opentrack package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/opentrack/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{ mkDerivation, lib, fetchFromGitHub, stdenv, cmake, pkg-config, ninja, qtbase
+, qttools, opencv4, procps, eigen, libXdmcp, libevdev, wineFull, fetchzip
+, makeDesktopItem, fetchurl, copyDesktopItems }:
+
+let
+  version = "2.3.13";
+
+  aruco =
+    import ./aruco.nix { inherit lib stdenv cmake fetchFromGitHub opencv4; };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/opentrack/default.nix:9]:
```suggestion
  aruco = import ./aruco.nix { };
```

File: pkgs/applications/misc/opentrack/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  # license.txt inside the zip file is MIT
+  xplaneSdk = fetchzip {
+    url =
+      "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK303.zip";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/opentrack/default.nix:14]:
```suggestion
    url = "https://developer.x-plane.com/wp-content/plugins/code-sample-generation/sample_templates/XPSDK303.zip";
```

File: pkgs/applications/misc/opentrack/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    "-DSDK_XPLANE=${xplaneSdk}"
+  ];
+
+  desktopItems = [
+    (makeDesktopItem rec {
+      name = "opentrack";
+      exec = "opentrack";
+      icon = fetchurl {
+        url =
+          "https://github.com/opentrack/opentrack/raw/opentrack-${version}/gui/images/opentrack.png";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/opentrack/default.nix:46]:
```suggestion
        url = "https://github.com/opentrack/opentrack/raw/opentrack-${version}/gui/images/opentrack.png";
```

File: pkgs/applications/misc/opentrack/default.nix (hunk 1 above)
Comment (zaninime) [reply pkgs/applications/misc/opentrack/default.nix:9]: Happy to apply the remaining suggestions, but I don't understand how this one can work. Could you please clarify?

**Comment by zaninime** [comment]
@SuperSandro2000 sorry to bother you, I know you're more than busy. Could you please clarify the above?

File: pkgs/applications/misc/opentrack/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/opentrack/default.nix:9]: If you don't list the attributes all required should be used IIRC.

File: pkgs/applications/misc/opentrack/default.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{ mkDerivation, lib, fetchFromGitHub, stdenv, cmake, pkg-config, ninja, qtbase
+, qttools, opencv4, procps, eigen, libXdmcp, libevdev, wineFull, fetchzip
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/opentrack/default.nix:2]:
```suggestion
, qttools, opencv4, procps, eigen, libXdmcp, libevdev, fetchzip
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/opentrack/default.nix (hunk 1 above)
Comment (zaninime) [reply pkgs/applications/misc/opentrack/default.nix]: That would be with `callPackage`. I changed it to that.

**Comment by zaninime** [comment]
@SuperSandro2000 I think I solved all the issues. Can you please have a look again?

File: pkgs/applications/misc/opentrack/aruco.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    runHook preInstall
+
+    mkdir -p $out/include/aruco
+    make install
+
+    # copy headers required by main package
+    cp $src/src/*.h $out/include/aruco
+
+    runHook postInstall
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/opentrack/aruco.nix:30]:
```suggestion
  preInstall = ''
    mkdir -p $out/include/aruco
  '';

  # copy headers required by main package
  postInstall = ''
    cp $src/src/*.h $out/include/aruco
  '';  
```

**Comment by zaninime** [comment]
@SuperSandro2000 suggestion applied.

**Pull request merged**